### PR TITLE
DOC: refresh "Why xarray" and shorten top-level description

### PR DESCRIPTION
**Pull request by shoyer**
This documentation revamp builds upon @rabernat's rewrite in #2430.

The main change is that the three paragraph description felt too long to me, so I moved the background paragraph on multi-dimensional arrays into the next section, on "Why xarray". I also ended up rewriting most of that page, and made a few adjustments to the FAQ and related projects pages.

### CODE REVIEW

File: doc/faq.rst [hunk 1]
```diff
@@ -39,9 +39,8 @@ if you were using Panels:
   xarray ``Dataset``.
 
 You can :ref:`read about switching from Panels to Xarray here <panel transition>`.
-Pandas gets a lot of things right, but scientific users need fully multi-
-dimensional data structures.
-
+Pandas gets a lot of things right, but scientific users (and many others) need
+fully multi-dimensional data structures.
```
Comment (rabernat) [inline doc/faq.rst]: Maybe a good place to explicitly mention deep learning libraries?

File: README.rst [hunk 2]
```diff
@@ -9,49 +9,47 @@ xarray: N-D labeled arrays and datasets
...
-Xarray also provides a large and growing library of functions for advanced
-analytics and visualization with these data structures.
 Xarray was inspired by and borrows heavily from pandas_, the popular data
 analysis package focused on labelled tabular data.
-Xarray can read and write data from most common labeled ND-array storage
-formats and is particularly tailored to working with netCDF_ files, which were
-the source of xarray's data model.
+It is particularly tailored to working with netCDF_ files, which were the
+source of xarray's data model, and integrates tightly with dask_ for parallel
+computing.
```
Comment (rabernat) [inline README.rst]: Great consolidation of the top-level description!

File: doc/why-xarray.rst [hunk 3]
```diff
@@ -1,11 +1,21 @@
...
+"tensors") are an essential part of computational science.
+They are encountered in a wide range of fields, including physics, astronomy,
+geoscience, bioinformatics, engineering, finance, and deep learning.
+In Python, NumPy_ provides the fundamental data structure and API for
+working with raw ND arrays.
+However, real-world datasets are usually more than just raw numbers;
+they have labels which encode information about how the array values map
+to locations in space, time, etc.
+
+Xarray doesn't just keep track of labels on arrays: it uses them to provide a
```
Comment (rabernat) [inline doc/why-xarray.rst]: nit: use a dash instead of a colon

**Comment by dcherian** [comment]
LGTM but I'll hold off on clicking merge. Apologies for merging too quickly the last time around.

**Pull request merged**